desktop/core/src/desktop/lib/botserver/views.py

```diff
@@ -143,14 +143,28 @@ def handle_on_link_shared(channel_id, message_ts, links, user_id):
       send_result_file(request, channel_id, message_ts, doc, 'xls')
 
 
-def slack_email_prefix(user_id):
+def slack_user_email_check(user_id):
+  """
+  Takes the user_id as input, gets the Slack user email.
+  Return the email prefix of those email whose email domain matches with host_domain
+
+  Eg: Assume host_domain be 'gethue.com',
+  If email is alice@gethue.com, then return 'alice'
+  If email is henry@example.com, then return None
+  
+  """
+
   try:
     slack_user = slack_client.users_info(user=user_id)
   except Exception as e:
     raise PopupException(_("Cannot find query owner in Slack"), detail=e)
-  
+
+  host_domain = '.'.join(conf.HTTP_HOST.get().split('.')[1:])
```

Comment:
For live editor like [demo.gethue.com](https://demo.gethue.com), `conf.HTTP_HOST.get()` will have value set as `demo.gethue.com` in hue.ini right??